File: backend/app/schemas/audit.py
"""审计日志响应契约。"""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AuditLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID | None
    username: str | None
    action: str
    target_type: str | None
    target_id: uuid.UUID | None
    detail: dict
    ip_address: str | None
    created_at: datetime


class AuditLogListResponse(BaseModel):
    items: list[AuditLogOut]
    page: int
    page_size: int
    total: int

File: frontend/src/types/audit.ts
export interface AuditLogRecord {
  id: string
  user_id: string | null
  username: string | null
  action: string
  target_type: string | null
  target_id: string | null
  detail: Record<string, unknown>
  ip_address: string | null
  success: boolean
  error_code: string | null
  request_id: string | null
  created_at: string
}

export interface AuditLogListResponse {
  items: AuditLogRecord[]
  page: number
  page_size: number
  total: number
}
